tests/test_zipfs.py

```diff
@@ -224,3 +226,38 @@ class TestOpener(unittest.TestCase):
     def test_not_writeable(self):
         with self.assertRaises(NotWriteable):
             open_fs("zip://foo.zip", writeable=True)
+
+
+class FSWithoutMtime(MemoryFS):
+    '''MemoryFS subclass that doesn't return details namespace
+    '''
+    def getinfo(self, path, namespaces):
+        if namespaces is not None:
+            namespaces = set(namespaces) - {'details'}
+        return super().getinfo(path, namespaces)
+
+
+class TestZipFSMtimeFallback(unittest.TestCase):
+    def setUp(self):
+        fh, self._temp_path = tempfile.mkstemp()
+        os.close(fh)
+
+    def tearDown(self):
+        os.remove(self._temp_path)
+
+    def test_no_mtime(self):
+        '''Fallback to current time when creating an archive of an fs that
+        doesn't support stat or details namespaces.
+        '''
+        src_fs = FSWithoutMtime()
+        with src_fs.open('test.txt', 'w') as f:
+            f.write('Hello World')
+
+        now = datetime.now()
```

Comment:
Another minor niggle, but I think this `now` variable would be better named `zip_write_datetime` or something?